File: src/main/java/tn/esprit/nafseyti/models/fiche_consultation.java
package tn.esprit.nafseyti.models;

import java.time.LocalDateTime;
import java.util.Objects;

public class fiche_consultation {
    private int id;
    private int rendezVousId;

    private String notes;
    private String problemePrincipal;
    private String diagnostic;
    private String recommandations;
    private String traitement;
    private int duree;
    private LocalDateTime createdAt;

    public fiche_consultation() {}

    public fiche_consultation(int id, int rendezVousId, String notes, String problemePrincipal, String diagnostic, String recommandations, String traitement, LocalDateTime createdAt) {
        this.id = id;
        this.rendezVousId = rendezVousId;
        this.notes = notes;
        this.problemePrincipal = problemePrincipal;
        this.diagnostic = diagnostic;
        this.recommandations = recommandations;
        this.traitement = traitement;
        this.createdAt = createdAt;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getRendezVousId() {
        return rendezVousId;
    }

    public void setRendezVousId(int rendezVousId) {
        this.rendezVousId = rendezVousId;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public String getProblemePrincipal() {
        return problemePrincipal;
    }

    public void setProblemePrincipal(String problemePrincipal) {
        this.problemePrincipal = problemePrincipal;
    }

    public String getDiagnostic() {
        return diagnostic;
    }

    public void setDiagnostic(String diagnostic) {
        this.diagnostic = diagnostic;
    }

    public String getRecommandations() {
        return recommandations;
    }

    public void setRecommandations(String recommandations) {
        this.recommandations = recommandations;
    }

    public String getTraitement() {
        return traitement;
    }

    public void setTraitement(String traitement) {
        this.traitement = traitement;
    }

    public int getDuree() {
        return duree;
    }

    public void setDuree(int duree) {
        this.duree = duree;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    @Override
    public String toString() {
        return "fiche_consultation{" +
                "id=" + id +
                ", rendezVousId=" + rendezVousId +
                ", notes='" + notes + '\'' +
                ", problemePrincipal='" + problemePrincipal + '\'' +
                ", diagnostic='" + diagnostic + '\'' +
                ", recommandations='" + recommandations + '\'' +
                ", traitement='" + traitement + '\'' +
                ", duree=" + duree +
                ", createdAt=" + createdAt +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof fiche_consultation that)) return false;
        return id == that.id && rendezVousId == that.rendezVousId && duree == that.duree && Objects.equals(notes, that.notes) && Objects.equals(problemePrincipal, that.problemePrincipal) && Objects.equals(diagnostic, that.diagnostic) && Objects.equals(recommandations, that.recommandations) && Objects.equals(traitement, that.traitement) && Objects.equals(createdAt, that.createdAt);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, rendezVousId, notes, problemePrincipal, diagnostic, recommandations, traitement, duree, createdAt);
    }
}

File: src/Entity/FicheConsultation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\FicheConsultationRepository;

#[ORM\Entity(repositoryClass: FicheConsultationRepository::class)]
#[ORM\Table(name: 'fiche_consultation')]
class FicheConsultation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function setId(int $id): self
    {
        $this->id = $id;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $rendez_vous_id = null;

    public function getRendez_vous_id(): ?int
    {
        return $this->rendez_vous_id;
    }

    public function setRendez_vous_id(?int $rendez_vous_id): self
    {
        $this->rendez_vous_id = $rendez_vous_id;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $notes = null;

    public function getNotes(): ?string
    {
        return $this->notes;
    }

    public function setNotes(?string $notes): self
    {
        $this->notes = $notes;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $probleme_principal = null;

    public function getProbleme_principal(): ?string
    {
        return $this->probleme_principal;
    }

    public function setProbleme_principal(?string $probleme_principal): self
    {
        $this->probleme_principal = $probleme_principal;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $diagnostic = null;

    public function getDiagnostic(): ?string
    {
        return $this->diagnostic;
    }

    public function setDiagnostic(?string $diagnostic): self
    {
        $this->diagnostic = $diagnostic;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $recommandations = null;

    public function getRecommandations(): ?string
    {
        return $this->recommandations;
    }

    public function setRecommandations(?string $recommandations): self
    {
        $this->recommandations = $recommandations;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $traitement = null;

    public function getTraitement(): ?string
    {
        return $this->traitement;
    }

    public function setTraitement(?string $traitement): self
    {
        $this->traitement = $traitement;
        return $this;
    }

    #[ORM\Column(type: 'datetime', nullable: false)]
    private ?\DateTimeInterface $created_at = null;

    public function getCreated_at(): ?\DateTimeInterface
    {
        return $this->created_at;
    }

    public function setCreated_at(\DateTimeInterface $created_at): self
    {
        $this->created_at = $created_at;
        return $this;
    }

}
